sonar-orchestrator/src/main/java/com/sonar/orchestrator/version/Version.java

```diff
@@ -19,43 +19,76 @@
  */
 package com.sonar.orchestrator.version;
 
+import org.jetbrains.annotations.Nullable;
+
 import java.util.Objects;
-import org.apache.commons.lang.StringUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Version implements Comparable<Version> {
 
+  private static final String GROUP_MAJOR        = "major";
```

Comment:
although for sure it is more readable, but unfortunately it does not follow our formatting guide, see https://github.com/SonarSource/sonar-developer-toolset